### DISCUSSION TITLE
CVE-2022-44268

### DISCUSSION THREAD

**Discussion by programmerarunkumar** [ImageMagick/ImageMagick#6031] [Help]
<policy domain="path" rights="none" pattern="@*" />
<policy domain="path" rights="none" pattern="|*" />

Team im having the above policy in my setup
 im having less than imagemagick version 7.0 

 will I face the (CVE-2022-44268)Information Disclosure Vulnerability of reading remote file issue ?

**Comment by urban-warrior** [comment]
ImageMagick strongly advises setting up a security policy that is suitable for your local environment.  Add this to your security policy:

```
<policy domain="path" rights="none" pattern="/etc/*"/>  <!-- don't read sensitive paths -->
```

With that policy, we get:

```
$ magick logo: -set profile /etc/passwd logo.png
magick: attempt to perform an operation not allowed by the security policy `/etc/passwd' @ error/blob.c/FileToBlob/1433.
```

You can be as draconian as you need.  Use `/*` as the path to prevent reading any file with an absolute path.  You can also protect against relative paths:

```
<policy domain="path" rights="none" pattern="../*"/>
```

See https://imagemagick.org/script/security-policy.php for a description of the ImageMagick security policy.

In the mean-time, this is an open dialog.  Feel free to post use cases we might have overlooked.

**Reply by programmerarunkumar** [reply]
<policy domain="path" rights="none" pattern="@*" />
<policy domain="path" rights="none" pattern="|*" />

**Reply by programmerarunkumar** [reply]
> ImageMagick strongly advises setting up a security policy that is suitable for your local environment. Add this to your security policy:
> 
> ```
> <policy domain="path" rights="none" pattern="/etc/*"/>  
> ```
> 
> With that policy, we get:
> 
> ```
> $ magick logo: -set profile /etc/passwd logo.png
> magick: attempt to perform an operation not allowed by the security policy `/etc/passwd' @ error/blob.c/FileToBlob/1433.
> ```
> 
> You can be as draconian as you need. Use `/*` as the path to prevent reading any file with an absolute path. You can also protect against relative paths:
> 
> ```
> <policy domain="path" rights="none" pattern="../*"/>
> ```
> 
> See https://imagemagick.org/script/security-policy.php for a description of the ImageMagick security policy.
> 
> In the mean-time, this is an open dialog. Feel free to post use cases we might have overlooked.

`<policy domain="path" rights="none" pattern="@*" />
<policy domain="path" rights="none" pattern="|*" />`

Im having this entry will i face issue ?

**Comment by urban-warrior** [comment]
You can validate your policy @ https://imagemagick-secevaluator.doyensec.com/.  You can trace its behavior by adding `-debug policy` to your command-line.

**Reply by fmw42** [reply]
Is this validator documented?  If not perhaps put it in the Misc section of https://imagemagick.org/script/sitemap.php or on the policy page at https://imagemagick.org/script/security-policy.php or wherever you think appropriate

**Comment by urban-warrior** [comment]
It's mentioned on the ImageMagick security policy web page.

**Reply by fmw42** [reply]
OK. Thanks. I overlooked it when looking on that page.
